### PR TITLE
Updated Democracy governance tutorial

### PR DESCRIPTION
**Pull request by ayushmishra2005**

### CODE REVIEW

File: docs/tutorials/tutorials-runtime-upgrade.md [hunk 1]
```diff
@@ -49,6 +48,46 @@ sidebar_label: Runtime Upgrade
 
 * Click the _Submit proposal_ button and sign the transaction
 
+![Submit Proposal](/img/tutorial/submit_proposal.png "Submit Proposal")
+
```
Comment (ltfschoen) [inline docs/tutorials/tutorials-runtime-upgrade.md:52]: a nice to have would be to mention what happens to the proposal if nobody seconds the proposal.
does it automatically get moved to the public referendum anyway?
and how many sessions or eras could go by before that happens, depending on the runtime configuration?

File: docs/tutorials/tutorials-runtime-upgrade.md [hunk 2]
```diff
@@ -49,6 +48,46 @@ sidebar_label: Runtime Upgrade
...
+![Second Proposal1](/img/tutorial/second_proposal1.png "Second Proposal1")
+
+![Second Proposal2](/img/tutorial/second_proposal2.png "Second Proposal2")
+
+![Second Proposal3](/img/tutorial/second_proposal3.png "Second Proposal3")
+
+### Step 5: Voting on the Proposal
+
+* The most seconded proposal moves to public referendum
```
Comment (ltfschoen) [inline docs/tutorials/tutorials-runtime-upgrade.md:67]: a nice to have would be to mention what happens to the proposal if nobody seconds the proposal.
does it automatically get moved to the public referendum for voting anyway?
and how many sessions or eras could go by before that happens, depending on the runtime configuration?
and to mention what happens if nobody votes on the referendum

File: docs/tutorials/tutorials-runtime-upgrade.md [hunk 3]
```diff
@@ -38,6 +36,7 @@ sidebar_label: Runtime Upgrade
 
 * Click "Submit Transaction" and then "Sign and Submit".
 
+![Submit Preimage](/img/tutorial/submit_preimage.png "Submit Preimage")
```
Comment (ltfschoen) [inline docs/tutorials/tutorials-runtime-upgrade.md:39]: i think it's worth mentioning that if they forgot to copy the preimage has, they can easily regenerate and copy it again by choosing the same extrinsic (but just not submitting it again)

File: docs/tutorials/tutorials-runtime-upgrade.md [hunk 4]
```diff
@@ -49,6 +48,46 @@ sidebar_label: Runtime Upgrade
...
+
+![Second Proposal3](/img/tutorial/second_proposal3.png "Second Proposal3")
+
+### Step 5: Voting on the Proposal
+
+* The most seconded proposal moves to public referendum
+
+* Once in referendum, token holders vote "Aye" or "Nay" on the proposal by locking tokens. Two factors account the vote weight: amount locked and locking period
+
+* If the proposal passes, it is enacted after a certain amount of time
```
Comment (ltfschoen) [inline docs/tutorials/tutorials-runtime-upgrade.md:71]: a nice to have is to mention what happens if nobody votes on the referendum and it does not pass, and how they could re-submit the proposal (i.e. regenerate the same pre-image hash without submitting it, and create a new proposal)

File: docs/tutorials/tutorials-runtime-upgrade.md [hunk 5]
```diff
@@ -49,6 +48,46 @@ sidebar_label: Runtime Upgrade
...
+
+![Second Proposal2](/img/tutorial/second_proposal2.png "Second Proposal2")
+
+![Second Proposal3](/img/tutorial/second_proposal3.png "Second Proposal3")
+
+### Step 5: Voting on the Proposal
+
+* The most seconded proposal moves to public referendum
+
+* Once in referendum, token holders vote "Aye" or "Nay" on the proposal by locking tokens. Two factors account the vote weight: amount locked and locking period
```
Comment (ltfschoen) [inline docs/tutorials/tutorials-runtime-upgrade.md:69]:
```suggestion
* Once in referendum, token holders vote "Aye" or "Nay" on the proposal by locking tokens. Two factors account the vote weight: amount locked (conviction) and locking period
```

**Review by ltfschoen — CHANGES_REQUESTED** [review]
this tutorial shows "how" to proposal a forkless runtime upgrade, but i think we need a second tutorial that explains to the community "why" they would want to propose one, and include some examples of what we'd encourage them to do.

for example, the "why" might be community might want the democracy process to run faster (similar to how we do it for debugging in development), so we'd explain how they'd have to modify the durations of parameters like `LaunchPeriod`, `VotingPeriod`, `FastTrackVotingPeriod`, `EnactmentPeriod`, and `CooloffPeriod`, in the runtime, before proceeding with the steps in this separate "How to propose a forkless runtime upgrade" tutorial.

so i think we need another step in this tutorial that's before the step "Step 1: Prepare WASM binary", and it should explain what the purpose of a forkless runtime upgrade is.

and it should provide different simple examples of code change scenarios with code snippets (what parts of the codebase they change) that they might do where they would need a forkless runtime upgrade, and examples of code changes that would cause further action like requiring network participants to restart their blockchain clients is required. 
it should mention how they'd need to modify the different components of the `RuntimeVersion` struct for each of the different scenarios prior to proceeding with the upgrade and provide links to where it is explained in detail here:
https://substrate.dev/rustdocs/latest/sc_executor/struct.RuntimeVersion.html#structfield.spec_version
https://substrate.dev/docs/en/tutorials/forkless-upgrade/sudo-upgrade#prepare-an-upgraded-

examples of code change scenarios could be:
* non-consensus-breaking optimizations
* adding/removing a module (pallet)
* changing a module's index (module ID)
* adding/removing a dispatchable
* changing a dispatchable's index (dispatch ID)
* adding/removing/changing a parameter
* changing user-level semantics
* changing runtime semantics
* changes outside the runtime folder (i.e. in the ./node folder)

it should also explain how to resolve exceptional situations where the node "client is the issue" and needs to be updated due to "breaking changes in the node between versions" where there have been "code changes outside your runtime folder (in your node folder)" and affecting how the chain spec is read, and how you would "update more than just the runtime, and install the new software", as mentioned here https://github.com/paritytech/subport/issues/208#issuecomment-891913804, and whether or not a "hard fork/spoon" or similar is necessary, what that would do that a forkless runtime upgrade can't do, and what steps that would involve (i.e. provide a link here where it explains what's involved in a "hard fork" including "requires node operators to stop their nodes and manually upgrade to the latest executable" https://substrate.dev/docs/en/knowledgebase/runtime/upgrades#forkless-runtime-upgrades). it should note that "forkless runtime upgrades do not require network participants to restart their blockchain clients" (see https://substrate.dev/docs/en/tutorials/forkless-upgrade/sudo-upgrade#start-the-template-node)

it should explain how the approach we've mentioned that the DAO would use to perform forkless runtime upgrades using the democracy pallet differs from using the sudo pallet (that they explain the the Substrate Developer Hub here https://substrate.dev/docs/en/tutorials/forkless-upgrade/) and why we want to remove Sudo (provide a link to the whitepaper here https://dev.datahighway.com/docs/whitepapers/whitepaper#financial-procedures)

File: docs/tutorials/tutorials-runtime-upgrade.md [hunk 6]
```diff
@@ -76,6 +80,8 @@ sidebar_label: Runtime Upgrade
 
 ![Vote3](/img/tutorial/vote3.png "Vote3")
 
+**Important** If referendum does not pass due to any reason(Either nobody votes or more `Nay` and less `Aye`), you can re-submit proposal using same preimage hash.
```
Comment (ltfschoen) [inline docs/tutorials/tutorials-runtime-upgrade.md:83]: could we please change `reason(Either` to `reason (either`

**Comment by sheenhx** [comment]
what is the balance requirement of a proposal? how much DHX?

File: docs/tutorials/tutorials-runtime-upgrade.md [hunk 7]
```diff
@@ -38,6 +38,8 @@ sidebar_label: Runtime Upgrade
 
 ![Submit Preimage](/img/tutorial/submit_preimage.png "Submit Preimage")
 
+**Important** If you forgot to copy the preimage hash, you can easily regenerate and copy it again by choosing the same extrinsic.(It won't allow to submit it again but you can get preimage hash.).
```
Comment (ltfschoen) [inline docs/tutorials/tutorials-runtime-upgrade.md:41]: could we please change `extrinsic.(It` to `extrinsic (it`, 
and change `hash.).` to `hash).`

**Review by ltfschoen — APPROVED** [review]
ready to merge, just a few nitpicks.
please create new Jira ticket for the "why" section and add link to the review comments

**Comment by ayushmishra2005** [comment]
> this tutorial shows "how" to proposal a forkless runtime upgrade, but i think we need a second tutorial that explains to the community "why" they would want to propose one, and include some examples of what we'd encourage them to do.
> 
> for example, the "why" might be community might want the democracy process to run faster (similar to how we do it for debugging in development), so we'd explain how they'd have to modify the durations of parameters like `LaunchPeriod`, `VotingPeriod`, `FastTrackVotingPeriod`, `EnactmentPeriod`, and `CooloffPeriod`, in the runtime, before proceeding with the steps in this separate "How to propose a forkless runtime upgrade" tutorial.
> 
> so i think we need another step in this tutorial that's before the step "Step 1: Prepare WASM binary", and it should explain what the purpose of a forkless runtime upgrade is.
> 
> and it should provide different simple examples of code change scenarios with code snippets (what parts of the codebase they change) that they might do where they would need a forkless runtime upgrade, and examples of code changes that would cause further action like requiring network participants to restart their blockchain clients is required.
> it should mention how they'd need to modify the different components of the `RuntimeVersion` struct for each of the different scenarios prior to proceeding with the upgrade and provide links to where it is explained in detail here (so the details of this part should be in the "how" tutorial):
> https://substrate.dev/rustdocs/latest/sc_executor/struct.RuntimeVersion.html#structfield.spec_version
> https://substrate.dev/docs/en/tutorials/forkless-upgrade/sudo-upgrade#prepare-an-upgraded-
> 
> examples of code change scenarios could be:
> 
> * non-consensus-breaking optimizations
> * adding/removing a module (pallet)
> * changing a module's index (module ID)
> * adding/removing a dispatchable
> * changing a dispatchable's index (dispatch ID)
> * adding/removing/changing a parameter
> * changing user-level semantics
> * changing runtime semantics
> * changes outside the runtime folder (i.e. in the ./node folder)
> 
> it should also explain how to resolve exceptional situations where the node "client is the issue" and needs to be updated due to "breaking changes in the node between versions" where there have been "code changes outside your runtime folder (in your node folder)" and affecting how the chain spec is read, and how you would "update more than just the runtime, and install the new software", as mentioned here [paritytech/subport#208 (comment)](https://github.com/paritytech/subport/issues/208#issuecomment-891913804), and whether or not a "hard fork/spoon" or similar is necessary, what that would do that a forkless runtime upgrade can't do, and what steps that would involve (i.e. provide a link here where it explains what's involved in a "hard fork" including "requires node operators to stop their nodes and manually upgrade to the latest executable" https://substrate.dev/docs/en/knowledgebase/runtime/upgrades#forkless-runtime-upgrades). it should note that "forkless runtime upgrades do not require network participants to restart their blockchain clients" (see https://substrate.dev/docs/en/tutorials/forkless-upgrade/sudo-upgrade#start-the-template-node)
> 
> it should explain how the approach we've mentioned that the DAO would use to perform forkless runtime upgrades using the democracy pallet differs from using the sudo pallet (that they explain the the Substrate Developer Hub here https://substrate.dev/docs/en/tutorials/forkless-upgrade/) and why we want to remove Sudo (provide a link to the whitepaper here https://dev.datahighway.com/docs/whitepapers/whitepaper#financial-procedures)

Added a ticket. https://mxc.atlassian.net/browse/MMD-1476

**Comment by ayushmishra2005** [comment]
> what is the balance requirement of a proposal? how much DHX?

@sheenhx While submitting the proposal, we set the locked balance. This is the number of tokens the proposer bonds with his/her proposal. The proposal with the most amount of tokens locked goes to referendum.

**Comment by ltfschoen** [comment]
> > what is the balance requirement of a proposal? how much DHX?
> 
> @sheenhx While submitting the proposal, we set the locked balance. This is the number of tokens the proposer bonds with his/her proposal. The proposal with the most amount of tokens locked goes to referendum.


@sheenhx the Democracy and Balance pallets have the following parameters whose values may be adjusted https://github.com/DataHighway-DHX/node/blob/master/runtime/src/lib.rs#L809

// Democracy pallet parameters
```
pub const MinimumDeposit: Balance = 100 * DOLLARS;
...
pub const PreimageByteDeposit: Balance = 1 * CENTS; 
```
// Balances pallet params
```
pub const ExistentialDeposit: Balance = 1 * DOLLARS;
```

where
1 DOLLAR 	 is 0.0001 DHX (1 milli DHX)
1 CENT   	         is 0.000001 DHX (1 micro DHX)
1 MILLICENTS is 0.000000001000000000 DHX
1 DHX = 10,000 DOLLARS = 1 MB, so 1 CENT per byte

to submit/register a pre-image for an upcoming proposal that does something like "staking > setValidatorCount > 11", a deposit is required and is calculated based on the the size of the pre-image in bytes using the per-byte fee, so in this example our pre-image would be ~3 bytes, and given a `PreimageByteDeposit` = 1 * CENTS, the calculated storage fee would be 3 micro DHX (3 cents).
there'd also be an addition tx fee to submit the pre-image of ~1.5 micro DHX.
so after the transaction, you can check the balance of the account that submitted it by going to https://polkadot.js.org/apps/?rpc=ws%3A%2F%2F127.0.0.1%3A9944#/chainstate, and querying Storage > system > account(AccountId): AccountInfo, and in this case it would display the following, since a tx fee was deducted from their free balance of 10 DHX, and 3 micro DHX was reserved for storage of the pre-image.
```
  data: {
    free: 9.9999 DHV,
    reserved: 3.0000 µDHV,
```

then to submit that proposal with that pre-image, an account would need to lock at least the `MinimumDeposit` amount shown above, which would be 0.01 DHX (which is `100 * DOLLARS`), and there'd also be an addition tx fee to submit the proposal of ~1.5 micro DHX.

then to second the proposal, the deposit required appears to be 0 DHX, but there's an tx fee associated of ~1.5 micro DHX.

so to submit a proposal to change the validator count using `setValidatorCount`, both the account that submits/registers the pre-image and the account that submits the proposal with that pre-image would need a minimum free balance (excluding locked/bonded/reserved) to pay for the transaction fees and to pay the storage fees without their free balance dropping below the existential deposit `ExistentialDeposit` amount mentioned above https://github.com/DataHighway-DHX/node/blob/master/runtime/src/lib.rs#L523

**Pull request merged**